src/iam/certhandler/certmodule.cpp

```diff
@@ -0,0 +1,394 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
+ * Copyright (C) 2023 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/iam/certmodule.hpp"
+#include "log.hpp"
+
+namespace aos {
+namespace iam {
+namespace certhandler {
+
+/***********************************************************************************************************************
+ * Public CertModule methods
+ **********************************************************************************************************************/
+
+CertModule::CertModule(const String& certType, const ModuleConfig& config)
+    : mCertType(certType)
+    , mModuleConfig(config)
+{
+}
+
+Error CertModule::Init(crypto::x509::ProviderItf& x509Provider, HSMItf& hsm, StorageItf& storage)
+{
+    mX509Provider = &x509Provider;
+    mHSM = &hsm;
+    mStorage = &storage;
+
+    if (mModuleConfig.mSkipValidation) {
+        LOG_WRN() << "Skip validation for certType: " << GetCertType();
+
+        return Error::Enum::eNone;
+    }
+
+    return SyncValidCerts();
+}
+
+const String& CertModule::GetCertType() const
+{
+    return mCertType;
+}
+
+RetWithError<CertInfo> CertModule::GetCertificate(const Array<uint8_t>& issuer, const Array<uint8_t>& serial)
+{
+    if (serial.IsEmpty()) {
+        const auto err = mStorage->GetCertsInfo(GetCertType(), mTmpCertsInStorage);
```

Comment:
Let's use just `auto` for `err`. It could be reused normally.

Reply:
fixed.